### ISSUE TITLE
Sphinx builds "succeed" despite errors

### ISSUE THREAD

**Issue by thijskh** [rst2pdf/rst2pdf#864]
I'm building my document with sphinx and rst2pdf 0.97.

When the package `fontconfig` is not installed, the build output shows an error, but the build succeeds with exit code 0 and a zero-byte document is produced. Probably the error should be caught and the build made to fail.

```
writing domjudge-team-manual... 
[ERROR] pdfbuilder.py:149 [Errno 2] No such file or directory: 'fc-match'
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/rst2pdf/pdfbuilder.py", line 146, in write
    docwriter.write(doctree, destination)
  File "/usr/lib/python3/dist-packages/docutils/writers/__init__.py", line 78, in write
    self.translate()
  File "/usr/lib/python3/dist-packages/rst2pdf/pdfbuilder.py", line 625, in translate
    createpdf.RstToPdf(sphinx=True,
  File "/usr/lib/python3/dist-packages/rst2pdf/createpdf.py", line 173, in __init__
    self.loadStyles(stylesheets)
  File "/usr/lib/python3/dist-packages/rst2pdf/createpdf.py", line 273, in loadStyles
    self.styles = sty.StyleSheet(styleSheets,
  File "/usr/lib/python3/dist-packages/rst2pdf/styles.py", line 358, in __init__
    fontList = findfonts.autoEmbed(style[key])
  File "/usr/lib/python3/dist-packages/rst2pdf/findfonts.py", line 322, in autoEmbed
    variants = findTTFont(fname)
  File "/usr/lib/python3/dist-packages/rst2pdf/findfonts.py", line 214, in findTTFont
    family = get_family(fname)
  File "/usr/lib/python3/dist-packages/rst2pdf/findfonts.py", line 178, in get_family
    data = make_string(subprocess.check_output(["fc-match", query]))
  File "/usr/lib/python3.8/subprocess.py", line 411, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.8/subprocess.py", line 489, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/usr/lib/python3.8/subprocess.py", line 854, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.8/subprocess.py", line 1702, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'fc-match'
FAILED
build succeeded.
```

**Comment by lornajane** [comment]
I'm sorry for the slow response here, we haven't been able to replicate the problem but I now think it is linked to a change in font cache file formats between fontconfig versions. Could you please tell us:
 - your operating system
 - the version of rst2pdf and how it was installed? E.g. Via snap, pypi, source

(anyone else with similar errors, please also add the same info. It really helps us!)

**Comment by stephenfin** [comment]
I'm able to reproduce this problem on a cloud Ubuntu 20.04 image, which comes with a pretty minimal amount of packages (I suspect the cloud Fedora images would have the same image). I don't think the real issue here is that running things without this package causes a failure or even  that we don't have this requirement documented (though that is an issue). Rather, it's that the build succeeds despite this failure. I recall seeing a similar issue about the return code for `rst2pdf` being 0 even when there's an issue. That's what we need to solve, IMO.

**Comment by akrabat** [comment]
See also: https://github.com/rst2pdf/rst2pdf/issues/886

**Comment by lornajane** [comment]
Can we revisit this now that we've merged #925? That change should make it more obvious when something has failed, but does it cover everything needed here?

**Comment by thijskh** [comment]
I have a hardish time reproducing the original issue. I am fine with closing this now that there's a concrete improvement in the code base. I'll be sure to let you know if I have (similar) issues in the future.
